File: backend/app/Models/Category.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

use App\Traits\BelongsToTenant;

class Category extends Model
{
    use BelongsToTenant;

    protected $fillable = ['tenant_id', 'name', 'nepali_name', 'image', 'order_index', 'is_active'];

    public function tenant(): BelongsTo
    {
        return $this->belongsTo(Tenant::class);
    }

    public function menuItems(): HasMany
    {
        return $this->hasMany(MenuItem::class);
    }
}

File: frontend/lib/models/category.dart
class Category {
  final int id;
  final int tenantId;
  final String name;
  final String? image;
  final int orderIndex;

  Category({
    required this.id,
    required this.tenantId,
    required this.name,
    this.image,
    required this.orderIndex,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      tenantId: json['tenant_id'],
      name: json['name'],
      image: json['image'],
      orderIndex: json['order_index'] ?? 0,
    );
  }
}
